Code/GraphMol/RascalMCES/RascalResult.h

```diff
@@ -0,0 +1,176 @@
+//
+// Copyright (C) David Cosgrove 2023
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+
+// A class to hold the results of a RASCAL MCES determination
+// between 2 molecules.  Contains the bonds and atoms that
+// correspond between the molecules, and also a SMARTS pattern
+// defining the MCES.
+//
+#include <RDGeneral/export.h>
+
+#ifndef RASCALRESULT_H
+#define RASCALRESULT_H
+
+#include <vector>
+
+#include <GraphMol/ROMol.h>
+
+namespace RDKit {
+
+namespace RascalMCES {
+
+class RDKIT_RASCALMCES_EXPORT RascalResult {
+ public:
+  RascalResult(const RDKit::ROMol &mol1, const RDKit::ROMol &mol2,
+               const std::vector<std::vector<int>> &adjMatrix1,
+               const std::vector<std::vector<int>> &adjMatrix2,
+               const std::vector<unsigned int> &clique,
+               const std::vector<std::pair<int, int>> &vtx_pairs, bool timedOut,
+               bool swapped, double tier1Sim, double tier2Sim,
+               bool ringMatchesRingOnly, bool singleLargestFrag,
+               int minFragSep);
+  // For when the tier[12]Sim didn't hit the threshold, but it
+  // might be of interest what the estimates of similarity were.
+  RascalResult(double tier1Sim, double tier2Sim);
+
+  RascalResult(const RascalResult &other);
+
+  RascalResult(RascalResult &&other) = default;
+
+  ~RascalResult() = default;
+
+  RascalResult &operator=(const RascalResult &other);
+
+  RascalResult &operator=(RascalResult &&other) = default;
+
+  // Cut the result down to the single largest fragment.  This is
+  // irrecoverably destructive.
+  void largestFragOnly();
+  void largestFragsOnly(unsigned int numFrags = 2);
+  void trimSmallFrags(unsigned int minFragSize = 3);
+
+  std::shared_ptr<RDKit::ROMol> getMol1() const { return d_mol1; };
+
+  std::shared_ptr<RDKit::ROMol> getMol2() const { return d_mol2; };
+
+  std::vector<std::pair<int, int>> getBondMatches() const {
+    return d_bondMatches;
+  }
+
+  std::vector<std::pair<int, int>> getAtomMatches() const {
+    return d_atomMatches;
+  }
+
+  int getNumFrags() const;
+
+  int getRingNonRingBondScore() const;
+
+  int getAtomMatchScore() const;
+
+  int getMaxDeltaAtomAtomDist() const;
```

Comment:
Do these really need to be in the public API?

Reply:
They are used by the now renamed resultCompare function.  I guess I could refactor that as operator< but I think that promises a bit more than the comparison actually delivers.
The score calculations are relatively expensive, so leaving them as member functions allows for lazy evaluation and saving the result.  Calculating them on the fly inside resultCompare slows the code down considerably.  I have an example where when using allBestMCESs and sorting the results it took about 25% of the run time just to do the sort when calculating the scores on the fly.